lib/CL/devices/remote/remote.c

```diff
@@ -1536,7 +1536,7 @@ pocl_remote_submit (_cl_command_node *node, cl_command_queue cq)
   POCL_INIT_COND (e_d->event_cond);
   e->data = (void *)e_d;
 
-  node->ready = 1;
+  node->state = POCL_COMMAND_READY;
   if (pocl_command_is_ready (node->sync.event.event))
```

Comment:
I think this should not be called at all here, just submit the command unconditionally to the remote.